packages/flutter/lib/src/gestures/resampler.dart

```diff
@@ -0,0 +1,424 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:collection';
+
+import 'events.dart';
+
+/// Class for pointer event resampling.
+///
+/// An instance of this class can be used to resample one sequence
+/// of pointer events. Multiple instances are expected to be used for
+/// multi-touch support. The sampling frequency and the sampling
+/// offset is determined by the caller.
+///
+/// This can be used get smooth touch event processing on devices
+/// without high frequency touch sensors. Resampling adds event
```

Comment:
Ditto: it would be nice to add cases like `or when the high input frequency is not a multiple of the display frequency (e.g., 120Hz input and 90Hz display)`.

Reply:
Done